library/std/src/io/stdio.rs

```diff
@@ -873,6 +875,49 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Reads a [`String`] from [standard input](Stdin). The trailing
+/// newline is stripped. Gives an error on EOF (end of file).
+///
+/// # Note
+///
+/// If you require more explicit control over capturing
+/// user input, see the [`Stdin::read_line`] method.
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(io_input)]
+/// use std::io;
+///
+/// fn main() -> io::Result<()> {
+///     print!("Enter name: ");
+///
+///     let name: String = io::read_line()?;
+///
+///     println!("Your name is {}!", name);
+///
+///     Ok(())
+/// }
+/// ```
+#[unstable(
+    feature = "io_input",
+    reason = "this function may be replaced with a more general mechanism",
+    issue = "none"
+)]
+pub fn read_line() -> Result<String> {
+    stdout().flush()?; // print!() does not flush :(
+    let mut input = String::new();
+    match stdin().read_line(&mut input)? {
+        0 => Err(Error::new(ErrorKind::UnexpectedEof, "input reached eof unexpectedly")),
+        _ => {
+            input.pop();
+            #[cfg(windows)]
+            input.pop();
+            Ok(input)
```

Comment:
This might pop non-newlines. When I end the input with ctrl-D instead of a newline on Unix, this drops the last character:

```
$ cargo r
input: hello^D^D
got: "hell"
```

Reply:
Shouldn't it return EOF error when this happens?

Reply:
And ignore the input? Why?

None of the other implementations (Rust's `BufRead::read_line`, C++'s `std::getline`, Python's `input()`, etc.) throw input away when there is no trailing newline. They all give the line of input, whether it ended in a newline or not.